common/opensearchconfigs.cpp

```diff
@@ -0,0 +1,84 @@
+/****************************************************************************
+**
+** Copyright (C) 2015
+** Contact: Siteshwar Vashisht <siteshwar@gmail.com>
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+#include <QDir>
+#include <QFile>
+#include <QStandardPaths>
+#include <QXmlStreamReader>
+#include "opensearchconfigs.h"
+
+OpenSearchConfigs *OpenSearchConfigs::openSearchConfigs = 0;
+
+OpenSearchConfigs::OpenSearchConfigs(QObject *parent):QObject(parent)
+{
+    m_userOpenSearchPath = QDir::homePath() + USER_OPENSEARCH_PATH;
+    m_openSearchPathList << QString(EMBEDLITE_CONTENT_PATH);
+    m_openSearchPathList << m_userOpenSearchPath;
+}
+
+const StringMap OpenSearchConfigs::parseOpenSearchConfigs()
+{
+    StringMap configs;
+
+    foreach(QString openSearchPath, m_openSearchPathList) {
+        QDir configDir(openSearchPath);
+        configDir.setSorting(QDir::Name);
+
+        foreach (QString fileName, configDir.entryList(QStringList("*.xml"))) {
+            QFile xmlFile(openSearchPath + fileName);
+            xmlFile.open(QIODevice::ReadOnly | QIODevice::Text);
+            QXmlStreamReader xml(&xmlFile);
+            QString searchEngine;
+
+            while (!xml.atEnd()) {
+                xml.readNext();
+                if (xml.isStartElement() && xml.name() == "ShortName") {
+                    xml.readNext();
+                    if (xml.isCharacters()) {
+                        searchEngine = xml.text().toString();
+                    }
+                }
+            }
+
+            if (!xml.hasError()) {
+                configs.insert(searchEngine, openSearchPath + fileName);
+            }
+
+            xmlFile.close();
+        }
+    }
+    return configs;
+}
+
+OpenSearchConfigs* OpenSearchConfigs::getInstance()
+{
+    if (!openSearchConfigs) {
+        openSearchConfigs = new OpenSearchConfigs();
+    }
+    return openSearchConfigs;
+}
+
+const QStringList OpenSearchConfigs::getSearchEngineList()
+{
+    // Return names of search engines
+    return getInstance()->parseOpenSearchConfigs().keys();
+}
+
+QString OpenSearchConfigs::userOpenSearchPath()
```

Comment:
Future has proved to be unpredictable enough, so it's better to remove the code that solves the problem which doesn't exist yet :)